### ISSUE TITLE
vec: clear

### ISSUE THREAD

**Issue by deltheil** [torch/tds#15]
Any reason to prevent removing all elements from the vector via `vec:resize(0)`? (see [this assert](https://github.com/torch/tds/blob/fb239ac0da2483a5fb86211ba80fb16f7a1243b2/vec.lua#L44)). Of course, as a workaround, one can do: `vec:resize(1); vec:remove(1)`.

Let me know what you think and I can send a PR (if there is any interest). Other possibilty: introduce a `clear` API:

```diff
diff --git a/vec.lua b/vec.lua
index b4aad33..a002028 100644
--- a/vec.lua
+++ b/vec.lua
@@ -45,6 +45,11 @@ function mt:resize(size)
    C.tds_vec_resize(self, size)
 end
 
+function mt:clear()
+   assert(self)
+   C.tds_vec_resize(self, 0)
+end
+
 if pcall(require, 'torch') then
    function mt:concatstorage(sep, i, j)
       i = i or 1
```

**Comment by soumith** [comment]
i dont see any reason to prevent this really!

**Comment by deltheil** [comment]
Allright! Will send a PR just in case.

**Comment by deltheil** [comment]
Oops it was a little bit too fast. I changed by mistake the assert message: `must be a strictly positive number` that should remain `must be a positive number`. Can you revert or force push? Sorry about that.

**Comment by deltheil** [comment]
Fixed by #19.
